### PR TITLE
[WIP] Reverts attack animations

### PR DESCRIPTION
**Pull request by ghost**
Gives us somewhere proper to talk about this rather than stretching it between 2 different discords and the thread

- [x] Fine-tune, first time trying a manual revert
- [x] Make enough popcorn to watch ensuing shitstorm
- [x] Get repobanned

### CODE REVIEW

**Comment by ghost** [comment]
frinken secret repo

**Comment by RoyalSev** [comment]
God bless you 
(but tbh it should be put to a poll _reeeeeeeeeeeeee_ )

**Comment by kilozombie** [comment]
Don't revert, IMHO.
It was a shitty thing to merge behind people's backs with the April Fool's thing, but overall they are serviceable and they're what people asked for.
The consensus seemed to be a poll after they've been in for a week at least, so let's wait for that but leave this open

**Comment by Intigracy** [comment]
They don't even affect you since they were made with a toggle that's off by default, sans the bug with it defaulting on.

Just turn them off and forget they exist, everyone wins.

**Comment by ghost** [comment]
it's cool

**Comment by ghost** [comment]
Now what does the drama tag do that controversial does not?

**Comment by TheFriggenSawbones** [comment]
fuck

**Comment by CptWad** [comment]
imo just have the animations "flash" so instead of a large fist hovering over my character for 3 seconds after its a flash whenever I do something

**Comment by clusterfack** [comment]
Its up there for three tenths of a second, I did it that long because I found I couldn't make out a good portion of objects when it was two tenths of a second.

**Comment by CptWad** [comment]
So is what I saw when A guy was running around with a big fist over him a bug or something?

**Comment by clusterfack** [comment]
Yeah it was, I applied a fix about 20 minutes after someone told me about it so it should already be gone this round or the next

**Comment by CptWad** [comment]
👌

**Comment by Probe1** [comment]
The ol "Why didn't coders fix the issue no one reported"

**Comment by kilozombie** [comment]
I think one concern is that item animations are linked to the thing you're hitting so if the hit breaks the thing, then the item animation goes away.
This is only in a few circumstances though, and none that involve actual combat, so I don't think it's a huge deal.

**Comment by Shadowmech88** [comment]
I don't like the circumstances they were merged under, but there *was* a server poll and a decision to definitely merge them in at some point based on the poll. Reverting it due to the merge circumstances anyway would serve no purpose.

If this is regarding the animations themselves, I don't believe enough time has passed since they were merged for players to judge them and possibly change their poll answers, not to mention we haven't had a new poll.

**Comment by carshalash** [comment]
The ol why didn't coders test their shit.

**Comment by TheFriggenSawbones** [comment]
bzsygw8cmaawt_x

**Comment by probemetagrudges** [comment]
hi long time player dont trust probe to not grudge against me especially after what he did to pomf but also want to let my voice be heard and you guys are ignoring the thread

**Comment by Fruktlimpa** [comment]
I can't for the life of me understand why you coders are being so defensive about this when you are the ones who created this drama shitstorm by not making a proper PR in the first place.
If you would have made a PR that showed all the polls and whatnot, with clear intentions to what it was you were going to merge then the drama would be negligible.

Just admit you fucked up with this so we can put it past us, but I guess your pride won't allow that.

**Comment by IAGFPRMF** [comment]
It's too late. Just let the TDM basketball Americans feel right at home with this wacky meme feature. Now that we are on our way to being all about "muh feedback" 

Chat wheel when?

**Comment by IAGFPRMF** [comment]
And regarding the bugged animations, it was working perfectly fine until the serb got all shitter shattered. It's causing all sorts of shit to be fucky.

**Comment by SonixApache** [comment]
there's been a poll
there's been three github threads with over 300 comments
p l e a s e

it's literally pride versus pride; the pride of dumb users that can't be assed to look past a single PR to see the copious amounts of work already done versus the pride of the dumb coder(s) that can't be assed to just re-write a PR so it doesn't feed the shitters

can we just drop this now and be done with it

**Comment by carshalash** [comment]
Maybe if you people actually waited more than one day to self merge attack animations back when the poll took place this mess wouldn't have happened. Did the thought that peoples opinions of that fiasco would have soured their opinion on attack animations to begin with.

**Comment by ghost** [comment]
There doesn't need a revert if they fix the bug with your setting not saving, I think the majority of people that dislike them will be happy if they can turn them off and they stay off forever.

**Comment by ultraplen** [comment]
Fix the bugs so that all the whiners can toggle the anims off so that we can all move on.

**Comment by PJB3005** [comment]
We need to take the opportunity to fix the bug to switch the default over to enabled though.

**Comment by NigglyWiggly** [comment]
We can switch the default over to enabled only if we make it a rule that coders need to be active players as well, or else their PR's get instantly flagged as un-mergable

**Comment by Arthurdentist** [comment]
Can't believe this still exists yet everyone mistook my mancrowave PR for an April Fools joke.

(I actually have no opinion on this but reading comments is fun)

**Comment by jakmak6** [comment]
It was a shitty idea that the community was so against they had to force merge it for april fools to mask it as a april fools joke along with all the other hilarity they had. Problem is they removed all the other meme stuff except this. Attack animations were barley tested and riddled with bugs and the community has expressed extreme disdain from them. I understand you want them in but with the autism surrounding it the only way to have it without being jewish is to hold a server poll to see if the want to keep or remove the attack animations. If it was so bad it had to be a april fools joke just to get it in then maybe it isnt such a bad idea to consider removing or atleast getting a vote in on it

**Comment by Riekland** [comment]
Was it autism?

**Comment by ultraplen** [comment]
@jakmak6 
>It was a shitty idea 

Check your pr history my dude :^) (ad hominem alert)

>that the community was so against 

#12853 and supposedly a server poll says otherwise. The guys spamming the 4chin thread hourly about the anims are not the whole community. If you are going to claim that the community was against the animations pre-merge, please link us a poll or something. 

>barley tested

That's a bold statement. Where did you hear they were barely tested before being merged?

>riddled with bugs 

Fair, but they will probably be worked out soon. The primary complaint seems to be regarding the settings not saving (edit: already fixed), which is just the type of thing that can slip by even the best coders quality control tbh.

The merge was undeniably shady but that doesn't mean that the animations should be removed entirely. It feels like the people who disliked them pre-merge are now trying to use this coder-hate memetrain to have a feature accepted and liked by the majority of the community removed.

**Comment by PJB3005** [comment]
Settings not saving is already fixed.

@jakmak6 can you please shut up EVERYTHING you've just said there is blatantly and objectively wrong.

**Comment by Fruktlimpa** [comment]
@ultraplen People are upset because you coders are acting like you haven't done anything wrong and that the thread is just shitposting and are attacking you 4noraisins.
It doesn't help that you keep adding fuel to this fire by acting like the hottest shit this server has ever had. People also see the bias that the coders have for each other.

I'd also like to add that I don't actually care if attack animations are in, I just hate the way you handled this and continue to double-down on this bullshit instead of swallowing your petty pride and admitting you caused this shitstorm for no benefit.

Simply put, drop the fucking superiority complex and everything is fine.

**Comment by Intigracy** [comment]
http://ss13.moe/index.php/poll/51

https://github.com/vgstation-coders/vgstation13/pull/12853

the toggle works, you can disable them and they stop affecting you

And yes this could have been handled better, hindsight is 20/20.

**Comment by clusterfack** [comment]
@carshalash I didn't merge the PR, what are you talking about.

**Comment by ultraplen** [comment]
@Fruktlimpa 
Your words are directed at some vague "coder" entity, not quite realising that "the coders" are a fairly large group holding varying opinions - I recommend atleast getting an overview good enough not to make such broad statements before posting.

Me, I'm an absolute nobody as a coder and as a player, so I have no right to speak for anyone but myself so don't go thinking that my opinions reflect those of every "coder" but:

>acting like you haven't done anything wrong
>caused this shitstorm for no benefit.

I've already stated that the circumstances of the merge were shitty, but as Shadowmech said reverting would serve no purpose. Also, the animations would have been merged without any further vote anyways. A major component in the outrage seems to be that people don't realise/accept that the animations have already been _discussed_, _emojied_ and _voted on_ and that the against side lost.

>acting like the hottest shit this server has ever had. 
>People also see the bias that the coders have for each other.

I don't understand how you can expect people who chat and collaborate daily to not have a bias for eachother. Putting that aside, it is only rational to hold the opinion of someone who has put in countless hours of work into this server higher than that of someone who signed up for Github yesterday to shit on a PR. The "hardcore" guys who have put their blood, sweat and tears into this server and its code deserve respect for it and I can't honestly see who would be "hotter shit" than them on here. That being said, there have been alot of rude and unconstructive posters on both sides.

I'd also like to add that being upset is not a valid reason to make up a bunch of garbage like @jakmak6  did.

**Comment by Iamgoofball** [comment]
It has a toggle, why do you need to take this feature away from the people that use it if you personally don't like it?

Or is this about ethics and github journalism again

**Comment by Iamgoofball** [comment]
>barley tested

Stop revealing the secret farm subsidies we benefit from goddamn it

**Comment by PJB3005** [comment]
Yeah no a revert isn't happening period. If you want to do a discussion about the coder ivory tower and transparency reverting attack animations is not the way to do it.

#14573

**Reverting attack animations is not happening, but if you want to discuss muh ivory tower drama and transparency issues see the above issue.**

**Comment by d3athrow** [comment]
in favor of a reversion

**Comment by ghost** [comment]
top cluck strikes again

**Comment by yclatious** [comment]
All hail the overlod Pomf

**Comment by ComicIronic** [comment]
>in favor of a reversion

On what grounds

**Comment by Iamgoofball** [comment]
Remember when pomf outright said "I don't play anymore so I'm not going to weigh in on this" in the last revert

**Comment by ghost** [comment]
to check what people are saying rather than closing down discussion

@Iamgoofball pls

**Comment by coldcola** [comment]
did someone e-transfer pomf their monthly autismbux??

**Comment by ComicIronic** [comment]
What people are saying has been said. Other parts of what people are saying is untrue. We can't rehash old arguments forever - if you think attack animations are ugly, turn them off. If you want them, turn them on. If there's something genuinely new to bring to this, I haven't seen it. This decision was basically already made. We aren't Congress, and every point of action is not a new opportunity to shitfling.

**Comment by PJB3005** [comment]
@ComicIronic I love being on your side of a discussion because you're amazing with words.

**Comment by Blithering** [comment]
Ugly or not, someone with attack animations on has a tangible advantage over someone without them on, such as being able to see someone on the other side of a wall drilling or hulkbusting through or being able to see the difference between a conversation and a punch up on cameras. 
Personally, I agree they are a bit ugly and detract from muh emulsions, but I have them enabled because at the end of the day, they do their job in making it clearer what's being hit with what. Locking threads is't going to help resolve this, but neither is revert or riot mindset. Best I can suggest is leaving it for long enough for people to get a feel for it and then holding a poll on whether to keep them or not.

**Comment by Exxion** [comment]
So turn them on

**Comment by PJB3005** [comment]
If people dislike them and don't want them on, I'm all ears on feedback for how an animation can be done to not be ugly.

If you don't want animations no matter what and think SS13 must be a fucking MUD, you're an idiot.

**Comment by Blithering** [comment]
Perhaps a less obtrusive animation would be one similar to the one that you get when hit with a projectile?

**Comment by Huuuuuuuuuuugo** [comment]
delet them you hack fraaaaauds

**Comment by SonixApache** [comment]
@Blithering there's an animation for getting hit?

**Comment by Exxion** [comment]
Yeah. Little dust puff. Technically it's the animation for a projectile hitting something, but close enough.

**Comment by aceedex** [comment]
Attack animations look terrible. Shouldn't be here.

Quality and peer review not memes and merging ok? Praise jesus

**Comment by Riekland** [comment]
The branch of vgstation-coders/vgstation13 gots to be refreshed from dime to dime with the blood of goder tyrannasauruses :-DD:-DDD:-D

**Comment by aceedex** [comment]
As for some actual thoughts on the matter, the do look very ugly and seeing carp triple punch windows out is kinda jarring

I do like the idea of the projectile hit animation, but personally I never had the issue of seeing who got hit/was hitting someone, seeing as you not only have a text box that plainly yells you whodunit, but the attacker sprites turns to face the victim, both get covered in blood, there's different sound effects for different items  and also the screams of " TERREL KILLING ME ESWORDS"

**Comment by Exxion** [comment]
So turn them off
Actually, they default off, so just don't turn them on

But yeah, no reason other than effort to not add more choices of animation

**Comment by carshalash** [comment]
Wut

**Comment by ghost** [comment]
I opened the PR for discussion, as the original point of it not being reverted was 'nobody can be bothered to try and revert it'

Now the discussion's largely over or transferred elsewhere. I'll keep the revert code for a later date should it be needed, but for now, we get some proper terms down regarding how the collabs should be acting.

**Comment by ghost** [comment]
Attack animations won the revert though. It's currently 18-12 in favor of a revert. This PR makes it obvious that players don't want it completely and so it should be reverted.

**Comment by ComicIronic** [comment]
emojicracy is not actually how PRs work, you know.

**Comment by ghost** [comment]
Then how do PRs work?

**Comment by PJB3005** [comment]
Maintainers make an educated decision by evaluating both sides of the argument.

**Comment by ghost** [comment]
Would anything have swayed the opinion of the coders to revert attack animations?

**Comment by Exxion** [comment]
A poll demonstrating the players are against it.
Which, if you're right that everyone hates them, will come in... a day or so? Don't remember when the week started.

**Comment by SolariousGabriella** [comment]
but I thought only opinions on the github mattered? hmmm
